### PR TITLE
Add SHA1 implementation

### PR DESCRIPTION
**Pull request by saleemrashid**
Import of Aaron D. Gifford's SHA1 implementation. This is needed for calculation of OpenPGP fingerprints.

### CODE REVIEW

**Comment by prusnak** [comment]
gI don't really like that sha1 code is being added to sha2.c.

Can we extract common stuff (like macros) to sha.h and then have SHA1 code in sha1.c? Or this would add too much work with small benefit?

File: sha2.c [hunk 1]
```diff
@@ -458,7 +876,7 @@ void sha256_Update(SHA256_CTX* context, const sha2_byte *data, size_t len) {
 			data += freespace;
 #if BYTE_ORDER == LITTLE_ENDIAN
 			/* Convert TO host byte order */
-			for (int j = 0; j < 16; j++) {
+			for (int j = 0; j < (SHA1_BLOCK_LENGTH >> 2); j++) {
```
Comment (prusnak) [inline sha2.c]: Is this intended? (using SHA1 macro in Sha256 code)

**Comment by saleemrashid** [comment]
@prusnak I agree. However, I wanted to replicate Aaron D. Grifford's code as much as possible.

File: sha2.c (hunk 1 above)
Comment (saleemrashid) [inline sha2.c]: Mea culpa, let me rebase

**Comment by saleemrashid** [comment]
@prusnak Fixed

**Comment by prusnak** [comment]
Thanks for you contribution!

**Pull request merged**